apps/src/lab2/views/dialogs/GenericDialog.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+
+import Typography from '@cdo/apps/componentLibrary/typography';
+import commonI18n from '@cdo/locale';
+
+import {useDialogControl} from './DialogControlContext';
+import {closeDialogType} from './types';
+
+export type ButtonType = 'confirm' | 'cancel' | 'neutral';
+
+export type dialogCallback = (args?: unknown) => void;
+
+type GenericDialogTitleProps =
+  | {
+      title?: never;
+      titleComponent?: React.ReactNode;
+    }
+  | {
+      title?: string;
+      titleComponent?: never;
+    };
+
+type GenericDialogBodyProps =
+  | {
+      message?: never;
+      bodyComponent?: React.ReactNode;
+    }
+  | {
+      message?: string;
+      bodyComponent?: never;
+    };
+
```

Comment:
typing is strict to allow a `title` OR `titleComponent` but not both.

Likewise a `message` or `bodyComponent` but not both.